core/platform/lf_os_single_threaded_support.c

```diff
@@ -0,0 +1,34 @@
+/**
+ * @file lf_os_single_threaded_support.c
+ * @author Marten Lohstroh (marten@berkeley.edu)
+ * @brief Implementation of platform functions to ensure safe concurrent
+ * access to a critical section, which are unnecessary in an OS-supported
+ * single-threaded runtime and therefore are left blank.
+ * @note This file is only to be used in conjuction with an OS-supported
+ * single-threaded runtime. If threads are enabled, this file will fail
+ * to compile. If threads are needed, use a multi-threaded runtime instead.
+ * @copyright BSD 2-Clause License (see LICENSE.md)
+ */
+
+#if defined(_THREADS_H) || defined(_PTHREAD_H)
+    #error Usage of threads in the single-threaded runtime is not safe.
+#endif
+
+int lf_critical_section_enter() {
+    // FIXME: Is this what we want? Dont we want to grab a mutex here? 
```

Comment:
I suppose on could manually add threading support while still using the single-threaded runtime, yes. But does it make sense to do this, or should we just say that this is not the right way. If we do take this stance, is there a foolproof way of detecting this and telling the programmer not to?

Reply:
I think the unthreaded runtime  + Linux is intended to be a very special case where you should not use other threads to schedule physical actions. I think the macro above would catch such a scenario. I think we can close this for now. I will remove the fixme